test/verify/test_fmod_mod.cpp

```diff
@@ -71,3 +71,4 @@ struct test_mod : verify_program<test_mod>
         return p;
     }
 };
+// TODO: check if requires FP8 test
```

Comment:
It appears we can do fmod with different types as long as fp8 follows is arithmetic
https://cplusplus.com/reference/type_traits/is_arithmetic/

https://cplusplus.com/reference/cmath/fmod/

Reply:
Yes it should work. 
But currently for some reason even `half_type` is breaking.  I'll have to check.

Reply:
https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/issues/2488

Not sure what is going on with `HalfType`.  Remainder seems to be returning Float vec. 

cc : @pfultz2

Reply:
Added test for `fp8` `fmod` and `mod`

Reply:
I think we observed this too when we were doing mod operator with the fmod flag, i think we did determine things were being cast to float instead of half, as part of the half ops since it was just reusing the std::fmod call still.